base/logging.jl

```diff
@@ -343,8 +343,9 @@ end
         # Try really hard to get the message to the logger, with
         # progressively less information.
         try
-            msg = "Exception while generating log record in module $_module at $filepath:$line"
-            handle_message(logger, Error, msg, _module, group, id, filepath, line; exception=err)
+            msg = "Exception while generating log record:\n"
+            msg *= join("  " .* split(sprint(showerror, err), "\n"), "\n")
+            handle_message(logger, Error, msg, _module, group, id, filepath, line)
```

Comment:
I still think this small part here should be reverted - it's in the wrong place because it's not the job of the log dispatch machinery to do formatting.  Formatting is the job of the installed Logger.  Doing formatting here is also inconsistent with the convention I've followed for exception logging elsewhere in `Base`.

If you want a replacement, it's quite easy to get equivalent but more general functionality by improving the way we print key value pairs in `SimpleLogger` when they're `Exception`s.

Reply:
That's what I did at first, but it felt weird to generate part of the message here and part of the message in `handle_message`. It is also nice not to have special casing inside `handle_message` based on `typeof(val)`.

Reply:
Perhaps the full try-catch should me moved to `handle_message` instead?